melodies_monet/plots/ozone_sonder_plots.py

```diff
@@ -0,0 +1,230 @@
+# Copyright (C) 2022 National Center for Atmospheric Research and National Oceanic and Atmospheric Administration
+# SPDX-License-Identifier: Apache-2.0
+#
+import os
+import monetio as mio
+import monet as monet
+import seaborn as sns
+from monet.util.tools import calc_8hr_rolling_max, calc_24hr_ave
+import xarray as xr
+import pandas as pd
+import numpy as np
+import cartopy.crs as ccrs
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+from numpy import corrcoef
+sns.set_context('paper')
+from monet.plots.taylordiagram import TaylorDiagram as td
+from matplotlib.colors import TwoSlopeNorm, ListedColormap, LinearSegmentedColormap
+from matplotlib.patches import Rectangle
+from matplotlib.ticker import FuncFormatter
+from monet.util.tools import get_epa_region_bounds as get_epa_bounds
+import math
+from ..plots import savefig
+from .surfplots import make_24hr_regulatory,calc_24hr_ave_v1,make_8hr_regulatory,calc_8hr_rolling_max_v1,calc_default_colors,new_color_map,map_projection,get_utcoffset,make_timeseries,make_taylor,calculate_boxplot,make_boxplot
+
+
+#Define ozone sonder, vertical single date plot
+def make_vertical_single_date(df,comb_bx,model_name_list,altitude_range,altitude_method,ozone_range,station_name,release_time,fill_color_list,plot_dict,fig_dict,text_dict):
+    ALT_sl = df['altitude']
+    O3_OBS = comb_bx[comb_bx.columns[0]].to_list()
+    len_combx = np.shape(comb_bx)[1]
+    O3_MODEL_ALL = []
+    for i in range(1,len_combx):
+        O3_MODEL = comb_bx[comb_bx.columns[i]].to_list()
+        O3_MODEL_ALL.append(O3_MODEL)
+    #release height info,get height of each release site to be substract    
+    df_height = pd.DataFrame({
+         'station':['Boulder, Colorado','Huntsville, Alabama','University of Rhode Island','Trinidad Head, California'],
+         'height':[1.743,0.203,0.021,0.046]
+         })
+    if altitude_method[0] == 'ground level':
+        height_value = df_height.loc[df_height['station']==station_name[0]].values[0][1]
+        ALT_gl = ALT_sl - height_value
+    #set default figure size
+    if fig_dict is not None:
+        f,ax = plt.subplots(**fig_dict)
+    else:
+        f,ax = plt.subplots(figsize=(8,8))
+    #set default text size
+    def_text = dict(fontsize=20)
+    if text_dict is not None:
+        text_kwargs = {**def_text, **text_dict}
+    else:
+        text_kwargs = def_text
+
```

Comment:
Having these plots replicated by altitude type seems redudent to me. Can you instead change this to:

if altitude_method[0] =='sea level':
      alt_p = ALT_sl
      alt_p_name = 'ALT-sea level (km)'
elif altitude_method[0] =='ground level':
      alt_p = ALT_gl
      alt_p_name = 'ALT-ground level (km)'

And then just have one plotting routine that uses these variables. You can probably even condense the calculations above into this logic too. This reduces the complexity of the code.

Reply:
this part is done!